tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -1155,6 +1156,28 @@ public:
    size_t GetAlignment() const override { return std::alignment_of<std::set<std::max_align_t>>(); }
 };
 
+/// The generic field for a std::map<KeyType, ValueType>
+class RMapField : public RProxiedCollectionField {
+private:
+   TClass *fItemClass;
+
+protected:
+   std::unique_ptr<Detail::RFieldBase> CloneImpl(std::string_view newName) const final;
+
+   std::size_t AppendImpl(const void *from) final;
+   void ReadGlobalImpl(NTupleSize_t globalIndex, void *to) final;
+
+public:
+   RMapField(std::string_view fieldName, std::string_view typeName, std::unique_ptr<Detail::RFieldBase> itemField);
```

Comment:
Hmm, this causes trouble with `CloneImpl` because when we call `Clone` on the subfield, a `RFieldBase` is returned (unless there's a way to cast it to an `RPairField` that I'm unaware of?)

Reply:
Right, I think we can do a `dynamic_cast` (for safety) to a `RPairField`. We know it must be a `RPairField`.